src/twisted/web/resource.py

```diff
@@ -409,6 +413,26 @@ def __init__(self, message="Sorry, resource is forbidden."):
 NoResource = _UnsafeNoResource
 ForbiddenResource = _UnsafeForbiddenResource
 
+# Explicitly deprecate the public aliases:
```

Comment:
This is the cause of the performance slowdown. Just, really shouldn't use `deprecatedModuleAttribute` if at all possible. That being said, doing "from twisted.web.resource import whatevs" instead of "from twisted.web import resource" in the right place would hopefully solve much of the slowdown.

Reply:
IIRC we hit this once before and that kind of fix addressed it almost completely?

Reply:
Yes, there were two perf-critical references via the module object (see [5f40096](https://github.com/twisted/twisted/pull/12240/commits/5f4009613c3db8e9348b2a07422b6a95ac518f06) and [7a0bfdc](https://github.com/twisted/twisted/pull/12240/commits/7a0bfdce6761579140d0e11b960716dcf8a23672)).